labonneboite/web/admin/views/office_admin_update.py

```diff
@@ -214,6 +226,53 @@ def create_form(self):
 
     def validate_form(self, form):
         is_valid = super(OfficeAdminUpdateModelView, self).validate_form(form)
+
+        # All sirets must be well formed
+        sirets = OfficeAdminUpdate.as_list(form.data['siret'])
+        only_one_siret = len(sirets) == 1
+
+        if is_valid and not sirets:
+            message = u"Le champs 'Sirets' est obligatoire. Veuillez le renseigner."
+            flash(message, 'error')
+            return False
+
+        if is_valid:
+            for siret in sirets:
+                if not is_siret(siret):
+                    message = u"Ce siret suivant n'est pas composé de 14 chiffres : {}".format(siret)
+                    flash(message, 'error')
+                    return False
+
+        # If only one siret, we valdate it
+        # If more than one siret : no siret validation
+        if is_valid and only_one_siret:
+            siret = sirets[0]
+            office = Office.query.filter_by(siret=siret).first()
+            if not office:
+                message = u"Le siret suivant n'est pas présent sur LBB: {}".format(siret)
+                flash(message, 'error')
+                return False
+
+        if is_valid:
+            for siret in sirets:
+                if 'id' in request.args:
+                    # Avoid conflict with himself if update by adding id != request.args['id']
```

Comment:
itself - object is not people :)

Reply:
or do you mean himself - the backoffice user?

Reply:
No it's itself ^^